src/odemis/acq/path.py

```diff
@@ -566,30 +567,27 @@ def _doSetPath(self, path, detector):
                                 # if there is a grating stored use this one
                                 # otherwise find the non-mirror grating
                                 if axis in self._stored:
```

Comment:
I think this is broken and it should change to `if (comp_role, axis) in self._stored`